plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -101,14 +122,53 @@ public DefaultThriftMetastoreClientFactory(
                 config.getConnectTimeout(),
                 config.getReadTimeout(),
                 metastoreAuthentication,
-                nodeManager.getCurrentNode().getHost());
+                nodeManager.getCurrentNode().getHost(),
+                buildThriftHttpContext(httpMetastoreConfig));
```

Comment:
No need to call `buildThriftHttpContext` method when httpMetastoreConfig is empty. e.g.
```java
httpMetastoreConfig.map(DefaultThriftMetastoreClientFactory::buildThriftHttpContext)
...
    @VisibleForTesting
    public static ThriftHttpContext buildThriftHttpContext(ThriftHttpMetastoreConfig config)
    {
        return new ThriftHttpContext(config.getHttpBearerToken(), config.getAdditionalHeaders());
    }
```

Reply:
Done.